presto-main/src/main/java/com/facebook/presto/sql/planner/DesugarAtTimeZoneRewriter.java

```diff
@@ -57,39 +53,25 @@ public static Expression rewrite(Expression expression, Session session, Metadat
         if (expression instanceof SymbolReference) {
             return expression;
         }
-        Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypes(session, metadata, sqlParser, symbolAllocator.getTypes(), expression, emptyList(), WarningCollector.NOOP);
-
-        return rewrite(expression, expressionTypes);
+        return new AnalyzedExpressionRewriter(session, metadata, sqlParser, symbolAllocator.getTypes()).rewriteWith(new Visitor(), expression);
     }
 
     private static class Visitor
-            extends ExpressionRewriter<Void>
+            extends ExpressionRewriter<Map<NodeRef<Expression>, Type>>
```

Comment:
hmmm.. should this `expressionType` be served as context? Let's talk in person.

Reply:
As discussed offline, will update to use a `RewriterProvider`